src/settings/settinggroup.cpp

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "settings/settinggroup.h"
+
+#include "leakdetector.h"
+#include "settingfactory.h"
+#include "settingsmanager.h"
+
+SettingGroup::SettingGroup(const QString& groupKey, bool removeWhenReset,
+                           bool sensitiveSetting, QStringList acceptedKeys)
+    : m_groupKey(groupKey),
+      m_sensitiveSetting(sensitiveSetting),
+      m_removeWhenReset(removeWhenReset),
+      m_acceptedKeys(acceptedKeys) {
+  MZ_COUNT_CTOR(Setting);
+
+  // Group settings are dynamic, therefore we need to load from memory all
+  // settings that exist under this group prefix in order to emit change signals
+  // when they change.
+  auto sb = SettingsManager::instance();
```

Comment:
Nit: 'sb' needs to be changed to match the new type name. Also, consider using a more descriptive name.